#170947441 return trip request

blaise82 opened the pull request:
### What does this PR do?

- validate return trip data
- add a return to type enum attribute
- add a return to type enum attribute to the model
- write tests

### How should this be manually tested?


### How should this be manually tested?
- clone this repo using `git clone https://github.com/andela/octopus-bn-backend`
- run `cd octopus-bn-backend` to move to the directory
- run `npm install` to install all required dependencies
- run `npm run dev` to start the development server
- open postman POST localhost:3000/api/v1/trips/request
- add the following body data example 
```
{
    "type": "string",
    "passportNumber": "string",
    "gender": "string",
    "fromCountry":"string",
    "fromCity":"string",
    "toCountry": "string",
    "toCity": "string",
    "departureDate": "string",
    "returnDate": "string",
    "reason": "the employee needs to attend a conference where the presence of a company representative is needed"
  }
```


### Stories associated with PR
<img width="1440" alt="Screen Shot 2020-02-19 at 14 00 44" src="https://user-images.githubusercontent.com/52036624/74832842-e7b9e500-5320-11ea-8988-d29b838efbab.png">

[#170947441](https://www.pivotaltracker.com/story/show/170947441)

### Screenshots if any
[finishes 170947441]

hakoemmy commented on src/database/models/travelRequests.js:
```diff
@@ -0,0 +1,23 @@
+  const travelRequests = sequelize.define('travelRequests', {
+    requestId: { type: DataTypes.UUID, allowNull: false },
+    userEmail: { type: DataTypes.STRING, allowNull: false },
+    type: DataTypes.ENUM('one way', 'return'),
```
@blaise82 
```suggestion
    type: DataTypes.ENUM('one-way', 'return'),
```
When you want to know one-way requests available in the database, you'll have to fetch a record and make a comparison so I think it would be easier if we have such ``one-way`` ENUM without space.

blaise82 replied:
@hakoemmy please refer the link https://sequelize.org/master/class/lib/data-types.js~ENUM.html#instance-constructor-constructor
sequelize.org giving an example of how ENUMs can be named

hakoemmy replied:
@blaise82 thanks for your response.

hakoemmy commented on src/routes/api/index.js:
```diff
@@ -14,6 +15,7 @@ router.use('/api/v1/auth', userRoutes);
 
 // set up auth routes v1
 router.use('/api/v1/auth', userRoutes);
+router.use('/api/v1/trip', tripsRouter);
```
```suggestion
router.use('/api/v1/trips', tripsRouter);
```
It's recommended to adhere [RESTful standards](https://restfulapi.net/resource-naming/)
that's to say; consider using plural for naming resources

hakoemmy replied:
@blaise82 What did you decide about resources naming?

blaise82 replied:
@hakoemmy  done

hakoemmy replied:
@blaise82  that's great

hakoemmy commented on src/routes/api/trips.routes.js:
```diff
@@ -0,0 +1,137 @@
+ *             properties:
+ *               type:
+ *                 type: string
+ *               passportNumber:
```
```suggestion
 *               passportNumber:
                          type: string
                          example: RK0884756
```
Would you please try to add an example for every key and value property? because it would be easier for us to have a clue on what to enter while we are testing this in ```swagger```

blaise82 replied:
@hakoemmy  great

hakoemmy replied:
@blaise82 Thank you for the response.

blaise82 replied:
@hakoemmy  done

blaise82 replied:
@hakoemmy  thanks for the feed back too

AbdoulNuru commented on src/controllers/tripsController.js:
```diff
@@ -0,0 +1,57 @@
+        fromCity, toCountry, toCity,
+        reason, departureDate, accommodation,
+        returnDate
+      } = req.body;
```
Document this method

blaise82 replied:
on it @AbdoulNuru

blaise82 replied:
@AbdoulNuru  Done

AbdoulNuru committed:
feat(trips): return trip request

- add return ENUM in types
- validate return date
- write tests

[Starts #170947441]
